include/NovelRTInterop/Maths/GeoMatrix4x4F_t.h

```diff
@@ -0,0 +1,346 @@
+    gm.x = value;
+  }
+
+  inline GeoVector4F_t GeoMatrix4x4F_getY(GeoMatrix4x4F_t& gm) {
+    return gm.y;
+  }
+
+  inline void GeoMatrix4x4F_setY(GeoMatrix4x4F_t& gm, const GeoVector4F_t& value) {
+    gm.y = value;
+  }
+
+  inline GeoVector4F_t GeoMatrix4x4F_getZ(GeoMatrix4x4F_t& gm) {
+    return gm.z;
+  }
+
+  inline void GeoMatrix4x4F_setZ(GeoMatrix4x4F_t& gm, const GeoVector4F_t& value) {
+    gm.z = value;
+  }
+
+  inline GeoVector4F_t GeoMatrix4x4F_getW(GeoMatrix4x4F_t& gm) {
+    return gm.w;
+  }
+
+  inline void GeoMatrix4x4F_setW(GeoMatrix4x4F_t& gm, const GeoVector4F_t& value) {
+    gm.w = value;
+  }
+
+  inline GeoMatrix4x4F_t GeoMatrix4x4F_getDefaultIdentity(GeoMatrix4x4F_t& gm) {
+    NovelRT::Maths::GeoMatrix4x4<float> matrix = NovelRT::Maths::GeoMatrix4x4(NovelRT::Maths::GeoVector4<float>(gm.x.x, gm.x.y, gm.x.z, gm.x.w),
+      NovelRT::Maths::GeoVector4<float>(gm.y.x, gm.y.y, gm.y.z, gm.y.w), NovelRT::Maths::GeoVector4<float>(gm.z.x, gm.z.y, gm.z.z, gm.z.w),
+      NovelRT::Maths::GeoVector4<float>(gm.w.x, gm.w.y, gm.w.z, gm.w.w));
+    NovelRT::Maths::GeoMatrix4x4<float> result = matrix.getDefaultIdentity();
+
+    GeoVector4F_t v1 = { result.getX().getX(), result.getX().getY(), result.getX().getZ(), result.getX().getW() };
+    GeoVector4F_t v2 = { result.getY().getX(), result.getY().getY(), result.getY().getZ(), result.getY().getW() };
+    GeoVector4F_t v3 = { result.getZ().getX(), result.getZ().getY(), result.getZ().getZ(), result.getZ().getW() };
+    GeoVector4F_t v4 = { result.getW().getX(), result.getW().getY(), result.getW().getZ(), result.getW().getW() };
+    return { v1, v2, v3, v4 };
+  }
+
+  inline bool GeoMatrix4x4F_equal(const GeoMatrix4x4F_t& first, const GeoMatrix4x4F_t& other) {
+    NovelRT::Maths::GeoVector4<float> first_vX = NovelRT::Maths::GeoVector4<float>(first.x.x, first.x.y, first.x.z, first.x.w);
+    NovelRT::Maths::GeoVector4<float> first_vY = NovelRT::Maths::GeoVector4<float>(first.y.x, first.y.y, first.y.z, first.y.w);
+    NovelRT::Maths::GeoVector4<float> first_vZ = NovelRT::Maths::GeoVector4<float>(first.z.x, first.z.y, first.z.z, first.z.w);
+    NovelRT::Maths::GeoVector4<float> first_vW = NovelRT::Maths::GeoVector4<float>(first.w.x, first.w.y, first.w.z, first.w.w);
+
+    NovelRT::Maths::GeoVector4<float> other_vX = NovelRT::Maths::GeoVector4<float>(other.x.x, other.x.y, other.x.z, other.x.w);
+    NovelRT::Maths::GeoVector4<float> other_vY = NovelRT::Maths::GeoVector4<float>(other.y.x, other.y.y, other.y.z, other.y.w);
+    NovelRT::Maths::GeoVector4<float> other_vZ = NovelRT::Maths::GeoVector4<float>(other.z.x, other.z.y, other.z.z, other.z.w);
+    NovelRT::Maths::GeoVector4<float> other_vW = NovelRT::Maths::GeoVector4<float>(other.w.x, other.w.y, other.w.z, other.w.w);
+
+    return NovelRT::Maths::GeoMatrix4x4<float>(first_vX, first_vY, first_vZ, first_vW) == NovelRT::Maths::GeoMatrix4x4<float>(other_vX, other_vY, other_vZ, other_vW);
+  }
```

Comment:
For `GeoMatrix4x4F_t`, this seems pretty painful and there's _a lot_ of it going on (mainly for all of the operator functions), and casting does not appear to work (as per the compiler ree'ing at me). Should I be handling this a separate way?
@RubyNova @tannergooding

Reply:
To clarify, what I feel is an issue here is that I have to create each specific `GeoVector4` first to build the `GeoMatrix4x4`, and I must do this every time we want to call `addVector`, `subtractVector`, etc. Add in the fact that I need to do this twice (once for the first Matrix, once for the seond Matrix), as well as taking the result and making it into the C-type (i.e. `GeoMatrix4x4<float>` to `GeoMatrix4x4F_t`) and you can probably see where I'm going with this.